### PR TITLE
Make all function annotations accessible from builtins complete

### PR DESCRIPTION
**Pull request by ddfisher**
Precursor to python/mypy/pull/1285

### CODE REVIEW

File: stdlib/2.7/__builtin__.pyi [hunk 1]
```diff
@@ -401,7 +403,7 @@ class bytearray(Sequence[int]):
     def translate(self, table: str) -> bytearray: ...
     def upper(self) -> bytearray: ...
     def zfill(self, width: int) -> bytearray: ...
-    @staticmethod
+    @classmethod
     def fromhex(self, x: str) -> bytearray: ...
```
Comment (gvanrossum) [inline stdlib/2.7/__builtin__.pyi]: In fact bytes.fromhex() is a static method. Also, the first arg should be named cls.

File: stdlib/2.7/__builtin__.pyi [hunk 2]
```diff
@@ -858,7 +860,7 @@ class file(BinaryIO):
     def __iter__(self) -> Iterator[str]: ...
     def read(self, n: int = ...) -> str: ...
     def __enter__(self) -> BinaryIO: ...
-    def __exit__(self, typ, exc, tb) -> bool: ...
+    def __exit__(self, t: type, exc: Any, tb: Any) -> bool: ...
```
Comment (gvanrossum) [inline stdlib/2.7/__builtin__.pyi]: The true signature of `__exit__` allows all three args to be optional. Also, the second could be restricted to `BaseException`.

File: stdlib/2.7/abc.pyi [hunk 3]
```diff
@@ -18,10 +18,10 @@ class ABCMeta(type):
     _abc_negative_cache = ...  # type: _weakrefset.WeakSet
     _abc_negative_cache_version = ...  # type: int
     _abc_registry = ...  # type: _weakrefset.WeakSet
-    def __init__(self, name, bases, namespace: Dict[Any, Any]) -> None: ...
-    def __instancecheck__(cls: "ABCMeta", instance) -> Any: ...
-    def __subclasscheck__(cls: "ABCMeta", subclass) -> Any: ...
-    def _dump_registry(cls: "ABCMeta", *args, **kwargs) -> None: ...
+    def __init__(self, name: str, bases: List[type], namespace: Dict[Any, Any]) -> None: ...
```
Comment (gvanrossum) [inline stdlib/2.7/abc.pyi]: `bases` should be a tuple.

File: stdlib/3/typing.pyi [hunk 4]
```diff
@@ -102,7 +102,7 @@ class Generator(Iterator[_T_co], Generic[_T_co, _T_contra, _V_co]):
     def send(self, value: _T_contra) -> _T_co:...
 
     @abstractmethod
-    def throw(self, typ: BaseException, val: Any=None, tb=None) -> None:...
+    def throw(self, typ: BaseException, val: Any = None, tb: Any = None) -> None:...
```
Comment (gvanrossum) [inline stdlib/3/typing.pyi]: The typ arg can also be an exception instance (then I think no value is allowed and I'm not sure about the tb).

File: stdlib/3/typing.pyi [hunk 5]
```diff
@@ -284,7 +284,7 @@ class IO(Iterable[AnyStr], Generic[AnyStr]):
     @abstractmethod
     def __enter__(self) -> 'IO[AnyStr]': ...
     @abstractmethod
-    def __exit__(self, type, value, traceback) -> bool: ...
+    def __exit__(self, t: type, value: Any, traceback: Any) -> bool: ...
```
Comment (gvanrossum) [inline stdlib/3/typing.pyi]: Same thing as before -- t is optional and value should be an `Optional[BaseException]`, I think.

File: stdlib/2.7/__builtin__.pyi (hunk 1 above)
Comment (ddfisher) [inline stdlib/2.7/__builtin__.pyi]: Changed to `@staticmethod` and removed first argument.

File: stdlib/2.7/__builtin__.pyi (hunk 2 above)
Comment (ddfisher) [inline stdlib/2.7/__builtin__.pyi]: Added some overloads; restricted second arg to `BaseException`.

File: stdlib/2.7/abc.pyi (hunk 3 above)
Comment (ddfisher) [inline stdlib/2.7/abc.pyi]: Changed.

File: stdlib/2.7/__builtin__.pyi (hunk 2 above)
Comment (ddfisher) [inline stdlib/2.7/__builtin__.pyi]: Realized that I didn't need overloads and added default args instead.

**Comment by gvanrossum** [comment]
Are you ready to push your fixes to the PR?

File: stdlib/3/typing.pyi (hunk 5 above)
Comment (ddfisher) [inline stdlib/3/typing.pyi]: Fixed.

**Comment by ddfisher** [comment]
Pushed.

**Comment by gvanrossum** [comment]
Test fail.

**Comment by ddfisher** [comment]
Fixed!

**Pull request merged**